planning.py

```diff
@@ -1,11 +1,57 @@
 """Planning (Chapters 10-11)
 """
 
+import copy
 import itertools
 from search import Node
-from utils import Expr, expr, first, FIFOQueue
+from utils import Expr, expr, first, FIFOQueue, partition
 from logic import FolKB
+from logic import fol_bc_and
+
+
+class PlanningKB:
+    """ A PlanningKB contains a set of Expr objects that are immutable and hashable.
+     With its goal clauses and its accompanying h function, the KB
+     can be used by the A* algorithm in its search Nodes. (search.py) """
+    def __init__(self, goals, initial_clauses=None):
+        if initial_clauses is None:
```

Comment:
I *think* you can do `if not initial_clauses`, but I am not 100% sure how it will work in this case. I'm pretty sure though this is how we generally have it in the repo, so if no issues will arise, I think it's best to change it.